docs/faq/missing-tailwind-classes.md

```diff
@@ -1,9 +1,78 @@
 # Why are my dynamic Tailwind classes not working?
 
-::: warning This answer is a stub.
-We are still working on writing the answers to the FAQ questions. The answer below is incomplete, but you may still find it useful.
-:::
+## How Tailwind works
 
-Because Tailwind can't tell that you're creating those classes dynamically.
+Tailwind will only include the CSS for a class if it finds that class somewhere in your code. You can read more about how it does that here:
 
-If you can't rewrite the code in such a way that Tailwind can detect them automatically then you'll need to add them to the safelist instead, <https://tailwindcss.com/docs/content-configuration#safelisting-classes>.
+<https://tailwindcss.com/docs/content-configuration#class-detection-in-depth>
+
+This happens as part of the build, using static analysis of the source code. It can't detect classes created at runtime.
+
+Tailwind isn't tied to a particular language or framework. It doesn't need to know whether you're using JavaScript or TypeScript, HTML or Vue SFCs. It processes all text in the file the same way, scanning for anything that looks like a Tailwind class name.
+
+## Dynamic class names
+
+Tailwind can't detect classes that are created dynamically using string manipulation. So it won't understand if you do something like this:
+
+```js
+const textColor = `text-${color}-500`
+```
+
+A dynamic class like this may still work, but only if something else adds the same class to the build. Dynamic classes can appear temperamental, with some values working while others don't. That's because the values that work are used elsewhere, which pulls them into the build.
+
+Tailwind recommends using a mapping for this kind of scenario:
+
+```js
+const themeColors = {
+  red: 'text-red-500',
+  green: 'text-green-500',
+  blue: 'text-blue-500'
+}
+
+const textColor = themeColors[color]
+```
+
+See <https://tailwindcss.com/docs/content-configuration#dynamic-class-names> for more details.
+
+You can also force Tailwind to include specific classes in the build using the `safelist` configuration option. See <https://tailwindcss.com/docs/content-configuration#safelisting-classes> for details.
+
+If you have too many different variants to list in a mapping or safelist, you may want to consider using a dynamic `style`, instead of Tailwind classes.
```

Comment:
```suggestion
If you have too many different variants to list in a mapping or safelist, you may want to consider using a dynamic `style`, instead of Tailwind classes. For example, if `color` can take an arbitrary value like `"#669900"`, instead of using dynamic class `:class="\'text-${color}\'"` you should use `:style="{ color }"` or `:style="\'color: ${color}\'"`.
```